rust/operator-binary/src/catalog/commons.rs

```diff
@@ -91,22 +93,55 @@ impl ExtendCatalogConfig for S3ConnectionInlineOrReference {
             .await
             .context(ConfigureS3Snafu)?;
 
-        catalog_config.add_property("hive.s3.endpoint", s3.endpoint().context(ConfigureS3Snafu)?);
-        catalog_config.add_property(
-            "hive.s3.path-style-access",
-            (s3.access_style == S3AccessStyle::Path).to_string(),
-        );
-
         let (volumes, mounts) = s3.volumes_and_mounts().context(ConfigureS3Snafu)?;
         catalog_config.volumes.extend(volumes);
         catalog_config.volume_mounts.extend(mounts);
 
+        if trino_version >= 469 {
+            // Since Trino 469, S3 native support has to be explicitly enabled
+            // unless using Legacy S3 support (which has been removed in 470).
+            catalog_config.add_property("fs.native-s3.enabled", "true");
+        }
+
+        let (endpoint_prop, region_prop, path_style_prop) = match trino_version {
+            ..=468 => (
+                "hive.s3.endpoint",
+                "hive.s3.region",
+                "hive.s3.path-style-access",
+            ),
+            469.. => ("s3.endpoint", "s3.region", "s3.path-style-access"),
+        };
+        catalog_config.add_property(endpoint_prop, s3.endpoint().context(ConfigureS3Snafu)?);
+
+        if let Some(region_name) = s3.region.name() {
+            catalog_config.add_property(region_prop, region_name);
+        }
+
+        catalog_config.add_property(
+            path_style_prop,
+            (s3.access_style == S3AccessStyle::Path).to_string(),
+        );
+
         if let Some((access_key, secret_key)) = s3.credentials_mount_paths() {
-            catalog_config.add_env_property_from_file("hive.s3.aws-access-key", access_key);
-            catalog_config.add_env_property_from_file("hive.s3.aws-secret-key", secret_key);
+            let (access_key_prop, secret_key_prop) = match trino_version {
+                ..=468 => ("hive.s3.aws-access-key", "hive.s3.aws-secret-key"),
+                469.. => ("s3.aws-access-key", "s3.aws-secret-key"),
+            };
+            catalog_config.add_env_property_from_file(access_key_prop, access_key);
+            catalog_config.add_env_property_from_file(secret_key_prop, secret_key);
         }
 
-        catalog_config.add_property("hive.s3.ssl.enabled", s3.tls.uses_tls().to_string());
+        match trino_version {
+            // Older trino versions allowed TLS to be optional
+            ..=468 => {
+                tracing::warn!("from Trino 460, TLS will be required for S3 connections");
```

Comment:
I think we should only log a warning in case TLS is not enabled (as only they will run into problems)

Reply:
Maybe I can do something like

```rust
            ..=468 if !tls_enabled => {
                tracing::warn!("from Trino 460, TLS will be required for S3 connections");
```

_Replacing `tls_enabled` with whatever applicable code to get what it describes_.

Reply:
Done in ac26e6f31aa69073d0dd7acf801f42683db03ca6